services/checks/checks.go

```diff
@@ -493,16 +505,32 @@ func (s *Service) findTargets(serviceType models.ServiceType) ([]target, error)
 
 	for _, service := range services {
 		e := s.db.InTransaction(func(tx *reform.TX) error {
-			a, err := models.FindPMMAgentsForService(s.db.Querier, service.ServiceID)
+			agents, err := models.FindPMMAgentsForService(s.db.Querier, service.ServiceID)
 			if err != nil {
 				return err
 			}
-			if len(a) == 0 {
+			if len(agents) == 0 {
 				return errors.New("no available pmm agents")
 			}
-			agent := a[0]
 
-			dsn, err := models.FindDSNByServiceIDandPMMAgentID(s.db.Querier, service.ServiceID, a[0].AgentID, "")
+			var agent *models.Agent
+			for _, a := range agents {
+				agentVersion, err := version.Parse(*a.Version)
+				if err != nil {
+					return errors.Wrap(err, "failed to checks agent version")
+				}
+
+				if !s.pmmVersion.Less(agentVersion) {
```

Comment:
I think we should have own version for each action type and looks like it's better to make this check here https://github.com/percona/pmm-managed/pull/400/files#diff-e0ff53b9498062b1544f8a3a40c5eb6bR384.

Reply:
`executeMongoDBChecks`

Reply:
So, I left version check here because we iterate each agent for each service to find proper version